### PR TITLE
Add zenchat to all domains pages

### PR DESCRIPTION
**Pull request by lsl**
Fixes https://github.com/Automattic/dotcom-forge/issues/3270

## Proposed Changes

* Adds zendesk chat bubble to all domains pages
* Reverts a change to `use-chat-status` that seemed partially responsible for presales chat not showing as often as it should.
* Skips availability check across the domain management UI and transfer flow.

Screenshot 2023-07-31 at 3 15 11 PM

## Todo:

We should create a domains context group for this, the wpcom group is not all that relevant:

Screenshot_2023-07-31_16-44-33

## Testing Instructions

* Visit http://calypso.localhost:3000/domains/manage, all pages under domains management should show the chat bubble.

After:
Screenshot (1)

### CODE REVIEW

File: client/components/data/domain-management/index.jsx [hunk 1]
```diff
@@ -61,11 +62,16 @@ class DomainManagementData extends Component {
...
+				<UsePresalesChat />
 			</div>
 		);
 	}
 }
 
+export function UsePresalesChat() {
+	usePresalesChat( 'wpcom' );
+}
+
```
Comment (lsl) [inline client/components/data/domain-management/index.jsx:74]: Is there a better way to do this?

**Comment by candy02058912** [comment]
The live chat bubble didn't show, is there anything else I should do other than check out to this branch?

File: client/components/data/domain-management/index.jsx (hunk 1 above)
Comment (klimeryk) [reply client/components/data/domain-management/index.jsx:74]: You can just call it directly, it's simply a hook: https://github.com/Automattic/wp-calypso/blob/32b0c450d155b58dafe38840f697bdbd42dd9a05/client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/index.tsx#L16

**Comment by klimeryk** [comment]
Make sure enhanced tracking protection (third party cookies, etc.) is disabled.
To verify your browser's settings are correct, check Help Center (the `?` icon top right corner, "Still need help?" and Live chat. If you see the contact form, all good. If you see this screen, you need to adjust your settings:
Screenshot 2023-07-31 at 16 49 15

Also, check this reply from @yoavf for other possible issues: p1690822162964229-slack-C4JPKKQBF

Basically:
> calypso.live is using the production zendesk environment, while calypso.local uses the dev environment.When using dev, agents needs to be logged in to the zendesk sandbox

**Comment by ebinnion** [comment]
I pushed a commit that bypasses the availability check for presales chat on the domains screens. In my testing, I was getting 503 errors on the sendpoint and `Not computer!` responses back from Zendesk when sandboxing `public-api.wordpress.com`.

File: client/lib/presales-chat/index.ts [hunk 2]
```diff
@@ -40,8 +36,7 @@ function getGroupName( keyType: KeyType ) {
 
 export function usePresalesChat( keyType: KeyType, enabled = true, skipAvailabilityCheck = false ) {
 	const isEnglishLocale = useIsEnglishLocale();
-	const { canConnectToZendesk } = useChatStatus();
-	const isEligibleForPresalesChat = enabled && isEnglishLocale && canConnectToZendesk;
+	const isEligibleForPresalesChat = enabled && isEnglishLocale;
```
Comment (klimeryk) [inline client/lib/presales-chat/index.ts]: I'd advise against this change. This was actually added in another PR related to this project: https://github.com/Automattic/wp-calypso/pull/79679 😅 

`canConnectToZendesk` is a cheap and fast check that tells us accurately if the user will be able to load the Zendesk's scripts or not (based on the security settings of their browser). There's no point in any other checks if this one returns false. Removing this check will not make Zendesk load magically - if the check failed, the script will not load anyway, so there will be no Zendesk launcher, etc.

If anything, I'd use the value of `canConnectToZendesk` to offer some other, backup support to the user. Either mentioning they need to adjust their browser's settings (but that's a bit meh and not exactly sets them up for best sales experience) or just show alternative resources that might help them in lieu of the pre-sales chat (I haven't checked the flow, so don't know if there's already something there or not).

Comment (ebinnion) [reply client/lib/presales-chat/index.ts]: Thanks. I wonder if the issue with the other PR then is that `canConnectToZendesk` returns false on loading.

https://github.com/Automattic/wp-calypso/blob/trunk/packages/help-center/src/hooks/use-chat-status.ts#L31

**Review by lupus2k — APPROVED** [review]
LGTM

File: client/components/data/domain-management/index.jsx (hunk 1 above)
Comment (candy02058912) [reply client/components/data/domain-management/index.jsx:74]: I think this was due to it being used in a class component.

Comment (klimeryk) [reply client/components/data/domain-management/index.jsx:74]: Ah, I see, I missed that!
